[ OEP 3903 ]: add pool cordon/uncordon feature proposal

tiagolobocastro opened the pull request:
## Description
Add OEP for mayastor pool cordon operations

## Motivation and Context
See the OEP

## Outcome
Please review the proposal

tiagolobocastro committed:
oep(3903): add pool cordon/uncordon feature proposal

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

Abhinandan-Purkait commented on designs/replicated-pv/mayastor/pool-cordon.md:
```diff
@@ -0,0 +1,144 @@
+this by:
+
+1. Cordon the pool
+2. Scale volumes with replicas on said pool by +1 replica
+3. Wait until the rebuilds complete
```
Does this imply the fact that once we scale down the replica on the cordoned pool would be preferred for removal?

tiagolobocastro replied:
Yes it does indeed

tiagolobocastro committed:
chore: add oep to commitlint

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

niladrih reviewed (approved):
Seems like a reasonable goal. Needs a corresponding how-to, I think.

tiagolobocastro committed:
ci(kind/deployer): add ip-family configuration

Adds cli switch for ip-family, allowing to more easily test clusters with
ipv4, ipv6 or dual ip stack.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>
